federation/executor.go

```diff
@@ -0,0 +1,89 @@
+package federation
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/samsarahq/thunder/graphql"
+	"github.com/samsarahq/thunder/graphql/introspection"
+)
+
+type ExecutorClient interface {
+	Execute(ctx context.Context, req *graphql.Query) ([]byte, error)
+}
+
+// Executor has a map of all the executor clients such that it can execute a
+// subquery on any of the federated servers.
+// The planner allows it to coordinate the subqueries being sent to the federated servers
+type Executor struct {
+	Executors map[string]ExecutorClient
+	planner   *Planner
+}
+
+func fetchSchema(ctx context.Context, e ExecutorClient) ([]byte, error) {
+	query, err := graphql.Parse(introspection.IntrospectionQuery, map[string]interface{}{})
+	if err != nil {
+		return nil, err
+	}
+
+	return e.Execute(ctx, query)
+}
+
+func NewExecutor(ctx context.Context, executors map[string]ExecutorClient) (*Executor, error) {
```

Comment:
Should we have a background process to regenerate the planner?

Reply:
We should, but this is going to happen in the proxy (a later thunder PR) that will regenerate the planner with updated schema